File: src/main/java/org/hwabeag/cashsystem/expansions/CashSystemExpansion.java
package org.hwabeag.cashsystem.expansions;

import me.clip.placeholderapi.expansion.PlaceholderExpansion;
import org.bukkit.OfflinePlayer;
import org.bukkit.configuration.file.FileConfiguration;
import org.hwabeag.cashsystem.CashSystem;
import org.hwabeag.cashsystem.config.ConfigManager;
import org.jetbrains.annotations.NotNull;

import java.util.Objects;

public class CashSystemExpansion extends PlaceholderExpansion {


    FileConfiguration PlayerConfig = ConfigManager.getConfig("player");
    private final CashSystem plugin; //

    public CashSystemExpansion(CashSystem plugin) {
        this.plugin = plugin;
    }

    @Override
    public boolean persist() {
        return true;
    }

    @Override
    public boolean canRegister() {
        return true;
    }

    @Override
    public @NotNull String getAuthor() {
        return plugin.getDescription().getAuthors().toString();
    }

    @Override
    public @NotNull String getIdentifier() {
        return "cash";
    }

    @Override
    public @NotNull String getVersion() {
        return plugin.getDescription().getVersion();
    }

    @Override
    public String onRequest(OfflinePlayer player, @NotNull String params) {
        if (params.equalsIgnoreCase("get")) { // %cash_get%
            String name = Objects.requireNonNull(player).getName();
            if (PlayerConfig.get(name + ".캐시") != null) {
                return String.valueOf(PlayerConfig.getInt(name + ".캐시"));
            }
            return "0";
        }
        return null; //
    }
}

File: src/main/kotlin/org/hwabeag/cashsystem/expansions/CashSystemExpansion.kt
package org.hwabeag.cashsystem.expansions

import me.clip.placeholderapi.expansion.PlaceholderExpansion
import org.bukkit.OfflinePlayer
import org.hwabeag.cashsystem.CashSystem

class CashSystemExpansion(private val plugin: CashSystem) : PlaceholderExpansion() {
    override fun persist(): Boolean = true

    override fun canRegister(): Boolean = true

    override fun getAuthor(): String = plugin.description.authors.joinToString(", ")

    override fun getIdentifier(): String = "cash"

    override fun getVersion(): String = plugin.description.version

    override fun onRequest(player: OfflinePlayer?, params: String): String? {
        if (!params.equals("get", ignoreCase = true) || player?.name == null) {
            return null
        }
        return CashSystem.cashRepository.getCash(player.name!!).toString()
    }
}

